### ISSUE TITLE
Some (certificate-related?) failures don't show up?

### ISSUE THREAD

**Issue by pepasflo** [mitmproxy/mitmproxy#2843]
First, thanks so much for working on mitmproxy! 👍

Recently, when debugging an iOS app, I noticed that a network failure showed up when using Charles proxy, but no traffic showed up at all in mitmproxy.

It looks like the failure was related to SSL -- I believe the iOS app was either doing cert pinning or otherwise trying to prevent proxies.  Charles was somehow able to show this failed request, whereas nothing showed up in the mitmproxy log.

If possible, it would be awesome to also have this ability in mitmproxy.

##### Steps to reproduce the problem:

1. Hrmm, sorry, I don't have a reliable way to reproduce this outside of my dev environment yet.

##### System information

```
$ mitmproxy --version
Mitmproxy version: 2.0.2 (release version) 
Python version: 3.6.2
Platform: Darwin-16.7.0-x86_64-i386-64bit
SSL version: OpenSSL 1.1.0f  25 May 2017
Mac version: 10.12.6 ('', '', '') x86_64
```

**Comment by mhils** [comment]
Not sure if this is the same issue, but I just noticed a similar problem when debugging #2840 where errors show up in mitmdump, but not mitmproxy. Repro:

```
curl -x localhost:8080 -k https://ondemand.icicibank.com/
```

**Comment by mhils** [comment]
This is not a regression from 2.0, so it's not a must-have for 3.0.

**Comment by mhils** [comment]
This should be fixed in 8.0.
